### PR TITLE
Add trough surrogate as unit model

### PR DESCRIPTION
**Pull request by kurbansitterley**
Converts work done for #20 to unit model format usable in flowsheet, associated data, and test files.

``solar_models > surrogate > trough`` contains:
- ``trough_surrogate.py`` is the unit model itself
- ``trough_surrogate.json`` is the surrogate model data
- ``test_trough_surrogate.py`` is the test file (wip)

``solar_models > surrogate > trough > data`` contains:
- ``pysam_run_trough.py`` used to generate data for surrogate
- ``training_trough_surrogate.py`` used to train surrogate
- ``trough_data.pkl`` data for surrogate
- other .jsons and weather file for PySAM

### CODE REVIEW

File: src/watertap_contrib/seto/analysis/net_metering/trough_LTMED.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    trough = m.fs.energy.trough
+    m.fs.energy.costing = EnergyCosting()
+    m.fs.energy.trough.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.energy.costing
+    )
+
+    m.fs.energy.costing.cost_process()
+    m.fs.sys_costing = SETOSystemCosting()
+
+    m.fs.energy.lt_med_heat_demand_constr = Constraint(
```
Comment (kurbansitterley) [inline src/watertap_contrib/seto/analysis/net_metering/trough_LTMED.py:191]: @adam-a-a here is the troublesome constraint!

**Review by adam-a-a — APPROVED** [review]
I have some suggested changes that shouldn't be too much of a lift. Looks good!

**Comment by Matthew-Boyd** [comment]
@kurbansitterley I think I need permissions to push to your repo--could you please enable that?

**Review by Matthew-Boyd — DISMISSED** [review]
I didn't spot any issues. I addressed Adam's comments in my similar PR [here](https://github.com/watertap-org/watertap-seto/pull/37/commits/7b580a23561093b912d966e00eb2fe66e4045725).

**Review by adam-a-a — APPROVED** [review]

**Review by zacharybinger — APPROVED** [review]

**Pull request merged**